### DISCUSSION TITLE
Add front matter support to `renderFile` in the Render plugin

### DISCUSSION THREAD

**Discussion by extrabright** [11ty/eleventy#4024] [Enhancement Queue]
### Describe the solution you'd like

I started using the `renderFile` option provided in the Render plugin: https://www.11ty.dev/docs/plugins/render/#renderfile. It works as expected, but when I use frontmatter in the rendered file, it doesn't seem to work.

It would be very helpful to have that and make it work 11tysh 💯 

```
{% renderFile "file.njk", { data } %}
```

In the rendered file:

```
---
title: Some title
description: ...
---

Content for included file 
```

### Describe alternatives you've considered

Before finding the Render plugin I created a custom tag for my 11ty projects which works similar, but I couldn't make it process the nunjucks shortcodes and filters inside the included file:

```
config.addNunjucksTag("inject", function(nunjucksEngine) {
   return new function() {
	this.tags = ["inject"];
	this.parse = function(parser, nodes) {
           const token = parser.nextToken()
	   const args = parser.parseSignature(null, true)

	    parser.advanceAfterBlockEnd(token.value)

	    return new nodes.CallExtensionAsync(this, 'run', args)
	}

	this.run = function(context, filePath, data, callback) {
		filePath = path.resolve('src/', filePath)

		// Allow usage without data passed to the tag
		if (callback == null) {
			callback = data
			data = null
		}

		fs.readFile(filePath, 'utf8', (err, str) => {

			if (err != null) return callback(err)

			var content = fm(str)
			var body = content.body

			let ret = nunjucksEngine.renderString(body, data, {
				path: filePath
			}, callback)

			ret = new nunjucksEngine.runtime.SafeString(ret);

			callback(null, ret);

		})
	};
   }();
});
```

**Comment by zachleat** [comment]
As noted in the docs https://www.11ty.dev/docs/plugins/render/#renderfile this is kinda by design (but only for implementation simplicity)

I’m definitely not opposed to adding this. I would also note that custom templates _do_ support front matter (just as evidence that it might be useful) https://www.11ty.dev/docs/languages/custom/

For complexity reasons I don’t think this would affect the data cascade at all—that data would be strictly localized to the renderFile instance.

Putting this one into the enhancement queue, make sure you upvote it!

> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by extrabright** [comment]
That sounds awesome!!! Hope to see it implemented soon 🔥
